### PR TITLE
Fix bug that prevents loading old models

### LINKED ISSUE #3357: How to load 3.8.3 gensim word2vec model by new version 4.2.0 gensim model
## Problem description

Hi! I have been using gensim 3.8.1 and want to use the latest version, gensim 4.2.0.

I have been able to load models made with older gensim versions, but when I changed to the latest version, I can no longer load them.
This is the error code.

```zsh
Exception has occurred: AttributeError
Can't get attribute 'Vocab' on <module 'gensim.models.word2vec' from '/PATH'>
```

I would like to know if there is some way to load older models.

## Versions

python 3.8.9
gensim 4.2.0

### PR DESCRIPTION
**Pull request by funasshi**
Fixes #3357.

# Problem
the model created by gensim 3.8.3 cannot be loaded by load function of gensim 4.2.0 and Attribute error occurs.

# How to solve 
import Vocab in word2vec.py

### CODE REVIEW

**Review by piskvorky — APPROVED** [review]

File: gensim/models/word2vec.py [hunk 1]
```diff
@@ -197,7 +197,7 @@
 import numpy as np
 
 from gensim.utils import keep_vocab_item, call_on_class_only, deprecated
-from gensim.models.keyedvectors import KeyedVectors, pseudorandom_weak_vector
+from gensim.models.keyedvectors import KeyedVectors, pseudorandom_weak_vector, Vocab
```
Comment (mpenkov) [inline gensim/models/word2vec.py:200]: Is this really enough? Can you please add a test that successfully loads an older gensim model?

**Review by mpenkov — CHANGES_REQUESTED** [review]

File: gensim/models/word2vec.py (hunk 1 above)
Comment (mpenkov) [reply gensim/models/word2vec.py:200]: Also, this import appears unused to flake8, so we should add a directive to ignore it.

File: gensim/models/word2vec.py [hunk 2]
```diff
@@ -200,6 +200,11 @@
 from gensim.models.keyedvectors import KeyedVectors, pseudorandom_weak_vector
 from gensim import utils, matutils
 
+#
+# This unused import is "somehow" required to fix #3357
```
Comment (piskvorky) [inline gensim/models/word2vec.py:204]: Explain the reason clearly – this comments not helpful.

**Review by piskvorky — CHANGES_REQUESTED** [review]

File: gensim/models/word2vec.py [hunk 3]
```diff
@@ -200,6 +200,11 @@
 from gensim.models.keyedvectors import KeyedVectors, pseudorandom_weak_vector
 from gensim import utils, matutils
 
+#
+# This import is required by pickle to load models stored by Gensim < 4.0, such as Gensim 3.8.3.
+#
```
Comment (piskvorky) [inline gensim/models/word2vec.py:205]: No need for this blank line – the comment above relates directly to the line below.

**Review by piskvorky — CHANGES_REQUESTED** [review]

**Review by piskvorky — APPROVED** [review]

**Comment by mpenkov** [comment]
Not sure why the changes are causing tests to consistently fail. Will need to look into it later.

**Comment by mpenkov** [comment]
Thank you @funasshi !

**Pull request merged**